logline.go

```diff
@@ -205,3 +205,38 @@ func (l *LogLine) Meta(key string) (interface{}, bool) {
 func (l *LogLine) Cache() *Cache {
 	return &l.cache
 }
+
+// Copy creates and returns a copy of the current log line.
+func (l *LogLine) Copy() Record {
+	// Copy metadata
+	md := make(Metadata)
+	for k, v := range l.meta {
+		md[k] = v
+	}
+
+	cpy := &LogLine{
+		cache:          l.cache,
+		wcnt:           l.wcnt,
+		meta:           md,
+		FieldSeparator: l.FieldSeparator,
+	}
+
+	if cpy.wcnt != 0 {
+		// If the log line has been modified, benchmarks have proven
+		// that it's more efficient to serialize and reparse to perform
+		// a copy (both in terms of time and allocation.
+		cpylen := len(l.data) + len(l.data)/2
+		text := l.ToText(make([]byte, 0, cpylen))
+		cpy.Parse(text, md)
```

Comment:
uhm, but at this point `cpy.wcnt` isn't wrong?

Reply:
if I'm not wrong after a parse it should be `0` but it's now `l.wcnt`

Reply:
You're right it should be 0 since all fields have been parsed and no fields have been written yet.
The line above where I set wcnt at LogLine creation is a leftover of a previous implementation and should be removed.

I'm surprised the tests didn't catch that though. I'm gonna write one that does.